### ISSUE TITLE
[data grid] column header is not properly draggable when IconButton is used in renderHeader

### ISSUE THREAD

**Issue by wilcoschoneveld** [mui/mui-x#18222]
### Steps to reproduce

Steps:
1. Open this link to live example: https://stackblitz.com/edit/nfbdiz9u?file=src%2FDemo.tsx
2. Try to drag the second column

### Current behavior

https://github.com/user-attachments/assets/1bd1baf4-3a98-4818-a649-1e5a28d09aca

you can only re-order the column when dragging via the `IconButton`

### Expected behavior

you should be able to re-order the column via the name
so the `IconButton` can be used for other things, e.g. show a dialog with information

### Context

Related issue which was closed incorrectly: https://github.com/mui/mui-x/issues/17133

We would like to show an info button in the column header (for this we have a small variant of the iconbutton) to allow the user to get more info about the column. We want IconButton because the button states give clear feedback to the user that the icon is clickable.

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Search keywords**: data grid column header ordering draggable custom renderheader

**Order ID**: 96051

**Comment by michelengelen** [comment]
Thanks for opening this.
I did a quick investigation and I think this is the culprit: https://github.com/mui/mui-x/blob/386a5b99e11f7e9220968167f9d1833285f00296/packages/x-data-grid/src/components/columnHeaders/GridGenericColumnHeaderItem.tsx#L89-L92

When the column header receives focus (is clicked on) and has no menu open we try to focus the first element in the tab cycle.

This breaks the drag behavior, since it moves the focus away from the draggable container, effectively stopping it.

I'll add this to the board for a deeper investigation and fix.
Thanks again for opening it! 👍🏼

**Comment by Danielvandervelden** [comment]
@michelengelen any update on this? I've actually encountered this issue today and took me a little while to figure out what was going on. Thanks!
